### PR TITLE
unnest deep loop

### PR DESCRIPTION
**Pull request by vienno**
Flat is better than nested. Instead of nesting `if` statements, check for the opposite early in the loop and `continue` if the opposite is the case.

### CODE REVIEW

**Comment by Lukasa** [comment]
Good idea @vienno!

If we're going to un-nest this, let's un-nest it the whole way. Want to break out the next nested couplet as well and use the same logic?

**Comment by miroli** [comment]
How about now @Lukasa ? I rewrote the third `if` statement as well, for consistency.

File: requests/cookies.py [hunk 1]
```diff
@@ -143,10 +143,13 @@ def remove_cookie_by_name(cookiejar, name, domain=None, path=None):
     """
     clearables = []
     for cookie in cookiejar:
-        if cookie.name == name:
-            if domain is None or domain == cookie.domain:
-                if path is None or path == cookie.path:
-                    clearables.append((cookie.domain, cookie.path, cookie.name))
+        if cookie.name != name:
+            continue
+        if domain and domain != cookie.domain:
```
Comment (Lukasa) [inline requests/cookies.py]: You changed this slightly. It needs to be `if domain is not None`.

File: requests/cookies.py [hunk 2]
```diff
@@ -143,10 +143,13 @@ def remove_cookie_by_name(cookiejar, name, domain=None, path=None):
...
+            continue
+        if path and path != cookie.path:
```
Comment (Lukasa) [inline requests/cookies.py]: Similarly, this needs to be `if path is not None`.

**Comment by Lukasa** [comment]
Thanks, much better! I have one small note, inline on the diff. =)

**Comment by miroli** [comment]
Of course! It's been updated.

**Comment by Lukasa** [comment]
Looks good to me @vienno! Would you like to add yourself to AUTHORS.rst?

**Comment by miroli** [comment]
Sure!

**Comment by Lukasa** [comment]
Thanks so much @vienno! :sparkles: :cake: :sparkles:

**Pull request merged**